spatial/r3/mat.go

```diff
@@ -197,3 +197,52 @@ func (m *Mat) Det() float64 {
 	detc := m.At(1, 0)*m.At(2, 1) - m.At(1, 1)*m.At(2, 0)
 	return a*deta - b*detb + c*detc
 }
+
+// Hessian sets the receiver to the hessian matrix of a spatial scalar field
+// evaluated at point p. Additional points evaluated to calculate hessian
+// are spaced at most tol away from p.
+func (m *Mat) Hessian(p, tol Vec, f func(Vec) float64) {
```

Comment:
I'd rename `tol` -> `step`.

Reply:
I'm going to reverse myself here now that the docs have been improved and I understand what is trying to be said better (this applies in the divergence PR too, but I'll add a comment there). Rather than `f` or `fn`, I think `field` documents it better, then the name parameter name doesn't need to be mentioned at all.

Reply:
And once you're at it, I'd simplify the information about evaluating in the neighborhood of p to just a mention of central differences, as in Divergence. And I'd completely avoid mentioning the mathematical assumptions on continuity - they are already implied by the fact that we're computing the Hessian. And what is "correct" when we're computing only an approximation (and in floating point)? They are not directly related to the implementation and so are out of scope. 

I'd write something like:
```
// Hessian sets the receiver to the Hessian matrix of the scalar field at the point p,
// approximated using finite differences with the given step sizes.
```